include/xproperty/xproperty.hpp

```diff
@@ -234,6 +237,15 @@ namespace xp
         return m_value;
     }
 
+
+    template <class T, class O, class D>
+    template <class Arg, class... Args>
+    inline auto xproperty<T, O, D>::operator()(Arg&& arg, Args&&... args) noexcept -> xp_owner_type
+    {
+        m_value = value_type(std::forward<Arg>(arg), std::forward<Args>(args)...);
+        return std::move(*owner());
```

Comment:
We should probably add a comment for specifying that it is meant for initialization, otherwise one may be surprised the the owner gets moved.

Reply:
```
inline auto xproperty<T, O, D>::operator()(Arg&& arg, Args&&... args) && noexcept -> xp_owner_type
```